docs/feature-flags.md

```diff
@@ -0,0 +1,47 @@
+# Feature Flags
+
+Features inside the `wc-admin` repository can be in various states of completeness. In addition to the development copy of `wc-admin`, feature plugin versions are bundled, and code is merged to WooCommerce core. To provide a way for improved control over how these features are released in these different environments, `wc-admin` has a system for feature flags.
+
+We currently support the following environments:
+
+| Environment | Description                                                                                                                                                            |
+|-------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| development | Development - All features should be enabled in development. These flags are also used in both JS and PHP tests. Ran using `npm start`.                                |
+| plugin      | Plugin - A packaged release of the featured plugin, for GitHub WordPress.org. Ran using `npm run-script build:release`. |                                    |
+| core        | Core - assets/files ready and stable enough for core merge. @todo No build process exists yet.
+
+
+## Adding a new flag
+
+Flags can be added to the files located in the `config/` directory. Make sure to add a flag for each environment and explicitly set the flag to false.
```

Comment:
I think the Calypso feature flags allow you to omit a flag from a config, and if it is not present it is considered that the feature is not enabled for that env. I believe initially you had to be explicit in adding a flag in all configs like it is setup here but changed during a re-write.

I'm fine with making this a requirement myself, but just figured I'd mention the path that happened on Calypso for reference.

Reply:
I just checked, and you actually can omit flags and still have it default to the feature being off (undefined).  So nothing will break if we forget to do this.

However, I would say let's keep the documentation line here. I like us having these be explicit in each env so you can quickly open the file and see what's up.

Reply:
Ah nice that it works that way. I'm fine with it as-is, just commenting about prior work on the same topic.